framework/src/timeintegrators/ActuallyExplicitEuler.C

```diff
@@ -96,6 +108,135 @@ ActuallyExplicitEuler::initialSetup()
 void
 ActuallyExplicitEuler::init()
 {
+  auto & es = _fe_problem.es();
+  auto & nonlinear_system = _fe_problem.getNonlinearSystemBase();
+  auto & libmesh_system = dynamic_cast<NonlinearImplicitSystem &>(nonlinear_system.system());
+  auto & system_matrix = *libmesh_system.matrix;
+
+  switch (_solve_type)
```

Comment:
What is all of this doing here in init()?  It doesn't make sense... and is a copy of the stuff below.

Reply:
We have u_dotdot in the system, therefore we need to define a initial u_dot (though default to zero). Once we have a initial u_dot, we need to compute an initial u_dotdot from the given u_dot to get the explicit algorithm started. I talked about this part in the write-up I attached above. Even if I change the formulation, I don't think we can avoid this initialization. There are quite a lot code duplication here, and I can try to simplify it a bit.